### PR TITLE
pod-utils: allow opting into censoring for secrets

### PR DESCRIPTION
**Pull request by stevekuznetsov**
Sidecar can now censor secrets from artifacts before upload. The new
option in decoration configuration allows users to opt into this - the
default behavior here is to mount all secrets that user code has access
to into the sidecar as well and have it censor all of them.

Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

/assign @alvaroaleman @cjwagner @petr-muller

### CODE REVIEW

File: prow/pod-utils/decorate/podspec.go [hunk 1]
```diff
@@ -705,10 +705,24 @@ func decorate(spec *coreapi.PodSpec, pj *prowapi.ProwJob, rawEnv map[string]stri
...
 	}
 
+	secretVolumes := sets.NewString()
+	for _, volume := range spec.Volumes {
+		if volume.VolumeSource.Secret != nil {
+			secretVolumes.Insert(volume.Name)
+		}
+	}
+	containsSecretData := func(volumeName string) bool {
+		if censor := pj.Spec.DecorationConfig.CensorSecrets; censor != nil && !*censor {
```
Comment (alvaroaleman) [inline prow/pod-utils/decorate/podspec.go:715]: Nit: A tad confusing to read, would be easier as:
```suggestion
		if censor := pj.Spec.DecorationConfig.CensorSecrets; censor == nil || !*censor {
```

**Review by alvaroaleman — COMMENTED** [review]
A test for this would be nice

**Comment by stevekuznetsov** [comment]
@alvaroaleman added tests

**Comment by alvaroaleman** [comment]
bazel still needs some appeasement :)

**Comment by stevekuznetsov** [comment]
doesn't it always?

**Comment by stevekuznetsov** [comment]
OK, this is ready for review & approval.

**Review by alvaroaleman — APPROVED** [review]

**Pull request merged**